src/d3a/models/market/__init__.py

```diff
@@ -102,11 +103,15 @@ def __init__(self, time_slot=None, bc=None, notification_listener=None, readonly
                 else TwoSidedMarketRedisEventSubscriber(self)
         setattr(self, RLOCK_MEMBER_NAME, RLock())
 
-    def _create_fee_handler(self, transfer_fees):
+    def _create_fee_handler(self, grid_fee_type, transfer_fees):
         if not transfer_fees:
             transfer_fees = TransferFees(grid_fee_percentage=0.0, transfer_fee_const=0.0)
-        if transfer_fees.transfer_fee_const is not None and transfer_fees.transfer_fee_const > 0.0:
-            self.fee_class = ConstantGridFees(transfer_fees.transfer_fee_const)
+        if grid_fee_type == 1:
+            if not (transfer_fees.transfer_fee_const is not None
+                    and transfer_fees.transfer_fee_const > 0.0):
+                self.fee_class = ConstantGridFees(0.0)
+            else:
+                self.fee_class = ConstantGridFees(transfer_fees.transfer_fee_const)
```

Comment:
Can you please explain this? grid_fee_type = 1 IS the constant grid fee type.

Reply:
Ahhh, no  I see it, it is a double "not"

Reply:
```suggestion
        if grid_fee_type == 1:
            if transfer_fees.transfer_fee_const is None or transfer_fees.transfer_fee_const <= 0.0:
                self.fee_class = ConstantGridFees(transfer_fees.transfer_fee_const)
            else:
                self.fee_class = ConstantGridFees(0.0)
```

Reply:
I will revert the double negation, however I am not merging the suggestion because the 2 blocks should be reversed. Pushing a separate change that fixes this.